backend/dataall/modules/datasets/indexers/dataset_catalog_indexer.py

```diff
@@ -22,5 +25,6 @@ def index(self, session) -> int:
         for dataset in all_datasets:
             tables = DatasetTableIndexer.upsert_all(session, dataset.datasetUri)
             folders = DatasetLocationIndexer.upsert_all(session, dataset_uri=dataset.datasetUri)
+            DatasetIndexer.upsert(session=session, dataset_uri=dataset.datasetUri)
             indexed += len(tables) + len(folders) + 1
-        return indexed
+        return indexed + len(all_datasets)
```

Comment:
I believe we add + 1 on L 29 to account for the dataset being indexed - either remove the +1 or remove the + len(datasets) on L30

Then update the test back to no changes

Reply:
The +1 was already there @noah-paige . Even before the `DatasetIndexer.upsert(session=session, dataset_uri=dataset.datasetUri)` was present. Was there for some other purpose ?

Reply:
Alright!. I will make the changes and update the code . Thanks @noah-paige